subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/GenericFileSystem.java

```diff
@@ -96,16 +97,22 @@ public GenericFileSystem(FileModeMutator chmod, FileModeAccessor stat, Symlink s
         this.symlink = symlink;
         this.chmod = chmod;
         canCreateSymbolicLink = symlink.isSymlinkSupported();
-        String content = generateUniqueContent();
-        File file = null;
-        try {
-            checkJavaIoTmpDirExists();
-            file = createFile(content);
-            caseSensitive = probeCaseSensitive(file, content);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        } finally {
-            FileUtils.deleteQuietly(file);
+    }
+
+    private void initializeCaseSensitive() {
+        if (caseSensitive == null) {
```

Comment:
The worst thing that could happen is that two threads do this work twice. They'll agree on the same answer in the end though.